aten/src/ATen/native/cuda/ForeachBinaryOpScalarList.cu

```diff
@@ -67,8 +68,53 @@ std::vector<Tensor> foreach_tensor_##NAME##_scalarlist_kernel_cuda(TensorList te
 }
 
 FOREACH_BINARY_OP_SCALARLIST(add, std::plus);
-FOREACH_BINARY_OP_SCALARLIST(sub, std::minus);
 FOREACH_BINARY_OP_SCALARLIST(mul, std::multiplies);
-FOREACH_BINARY_OP_SCALARLIST(div, std::divides);
+
+// In the case of division, integer inputs will result in float. 
+// Currently multi tensor apply can only return result of the same type as input.
+void foreach_tensor_div_scalarlist_kernel_cuda_(TensorList tensors, at::ArrayRef<Scalar> scalars) {
+    check_foreach_api_restrictions(tensors, scalars);
+    if (!can_use_fast_route(tensors, scalars, /*div_op*/ true)) {
+        return at::native::foreach_tensor_div_scalarlist_kernel_slow_(tensors, scalars);
+    }
+
+    foreach_binary_op_<std::divides>(tensors, scalars);
+}
+
+std::vector<Tensor> foreach_tensor_div_scalarlist_kernel_cuda(TensorList tensors, at::ArrayRef<Scalar> scalars) {
+    check_foreach_api_restrictions(tensors, scalars);
+    if (!can_use_fast_route(tensors, scalars, /*div_op*/ true)) {
+        return at::native::foreach_tensor_div_scalarlist_kernel_slow(tensors, scalars);
+    }
+
+    return foreach_binary_op<std::divides>(tensors, scalars);
+}
+
+// In the case of subtraction, we dont allow scalar to be boolean following the torch.sub logic
```

Comment:
It would be good to mention "This does not use FOREACH_BINARY_OP_SCALARLIST because ". It's easy to make the inference but just to clarify that this is a special case

Reply:
Done